final-project/manConstraintsJustification.tex

```diff
@@ -9,9 +9,7 @@ \chapter{Constraints \& Justification \\
 
 TODO: describe justification for why leader election will be too complex
 
-TODO: mention constraints imposed by the browser's websocket API (e.g. no custom HTTP headers)
 \section{Browser Websocket API Contraints}
 
 The Websocket API allows for one to open a bidirectional communication session between a browser and a server. The connection session stays open until the browser or server terminates it. 
-This allows for the client and server to send information to the other simultaneously. While useful, it does have constraints. 
-The Websocket API does not allow custom HTTP headers, which means that authorization has to be done after the connection request has been made.
\ No newline at end of file
+This allows for the client and server to send information to the other simultaneously. While useful, the Websocket API does not allow custom HTTP headers, which means that authorization has to be done after the connection request has been made \cite{HerokuWebSocket}.
```

Comment:
It's the _browser's_ API that doesn't allow you to add custom headers. For example, cookies _will_ be sent because cookies are handled by browsers by default. You could also cite this in addition: https://developer.mozilla.org/en-US/docs/Web/API/WebSocket

Reply:
okay thank you for the clarification!